Modules/Sources/BuildSettingsKit/BuildSettings+Preview.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+/// The container for Xcode previews.
+extension BuildSettings {
+    nonisolated(unsafe) static var preview = BuildSettings(
+        pushNotificationAppID: "xcpreview_push_notification_id",
+        appGroupName: "xcpreview_app_group_name",
+        appKeychainAccessGroup: "xcpreview_app_keychain_access_group"
+    )
+}
+
+extension BuildSettings {
+    /// Updates the preview settings for the lifetime of the given closure.
+    /// Reverts to the original settings when done.
+    @MainActor
+    public static func withSettings<T>(_ configure: (inout BuildSettings) -> Void, perform closure: () -> T) -> T {
```

Comment:
`BuildSettings.preview` isn't thread-safe, but we can live with that for the purposes of main-thread confined Xcode Previews.